src/main/java/io/confluent/idesidecar/restapi/connections/ConnectionState.java

```diff
@@ -65,14 +69,54 @@ public void setSpec(ConnectionSpec in) {
     this.spec = in;
   }
 
-  public Future<ConnectionStatus> getConnectionStatus() {
-    return Future.succeededFuture(ConnectionStatus.INITIAL_STATUS);
+  /**
+   * Obtain the most recently-obtained status of the connection.
+   *
+   * @return the connection status; never null
+   */
+  public ConnectionStatus getConnectionStatus() {
+    return this.cachedStatus.get();
   }
 
-  public ConnectionStatus getInitialStatus() {
+  /**
+   * Return the status for a newly-created connection. This method can be overridden by subclasses
+   * to provide a different initial status.
+   *
+   * @return the initial status; never null
+   */
+  protected ConnectionStatus getInitialStatus() {
     return ConnectionStatus.INITIAL_STATUS;
   }
 
+  /**
+   * Attempt to {@link #doRefreshConnectionStatus() refresh the connection status and
+   * update the {@link #getConnectionStatus() cached results}.
+   *
+   * <p>This method always calls {@link #doRefreshConnectionStatus()} and then on success updates
+   * the {@link #getConnectionStatus() cached connection status}.
+   *
+   * @return the future that will complete with the updated connection status
+   * @see #getConnectionStatus()
+   * @see #doRefreshConnectionStatus()
+   */
+  public final Future<ConnectionStatus> refreshConnectionStatus() {
+    // Always set the cached status when the future completes successfully
+    return doRefreshConnectionStatus().onSuccess(this.cachedStatus::set);
+  }
+
+  /**
+   * Refresh the connection status. By default, this simply returns a completed future with the
+   * {@link #getInitialStatus() initial status}. Subclasses should override this method to
+   * implement the actual connection status refresh logic.
+   *
+   * @return the future that will complete with the updated connection status
+   * @see #refreshConnectionStatus()
+   * @see #getConnectionStatus()
+   */
+  protected Future<ConnectionStatus> doRefreshConnectionStatus() {
+    return Future.succeededFuture(getInitialStatus());
+  }
+
```

Comment:
Because we don't want subclasses being responsible for updating the status, I chose to make the public `refreshConnectionStatus()` method _final_, and to move the overridable logic to the new protected `doRefreshConnectionStatus()` method.

I'm not sure about method names, and I was thinking that maybe `doRefreshConnectionStatus()` should be made abstract. WDYT?